leshan-core/src/main/java/org/eclipse/leshan/senml/SenMLPackSerDes.java

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     Boya Zhang - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.leshan.senml;
+
+import org.eclipse.leshan.core.model.ResourceModel.Type;
+import org.eclipse.leshan.util.Hex;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+
+public class SenMLPackSerDes {
+
+    public String serialize(SenMLPack pack) {
+        JsonArray jsonArray = new JsonArray();
+
+        for (int i = 0; i < pack.getRecords().size(); i++) {
+            JsonObject jsonObj = new JsonObject();
+
+            if (i == 0 && pack.getBaseName() != null) {
+                jsonObj.add("bn", pack.getBaseName());
+            }
+            if (i == 0 && pack.getBaseTime() != null) {
+                jsonObj.add("bt", pack.getBaseTime());
+            }
+
+            SenMLRecord records = pack.getRecords().get(i);
+            if (records.getName() != null && records.getName().length() > 0) {
+                jsonObj.add("n", records.getName());
+            }
+
+            if (records.getTime() != null) {
+                jsonObj.add("t", records.getTime());
+            }
+
+            Type type = records.getType();
+            if (type != null) {
+                switch (records.getType()) {
+                case FLOAT:
+                case INTEGER:
```

Comment:
reading the spec I didn't see INTEGER type :
https://tools.ietf.org/html/rfc8428#section-4.3
http://www.openmobilealliance.org/release/LightweightM2M/V1_1-20180710-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1-20180710-A.html#7-4-4-0-744-JSON

Reply:
By referring above link, my understanding float value attribute of SenML record can be used to represent Integer, Float or Time Resource data type. Time Resource data type, the value shall be the difference, measured in seconds(not milliseconds), between the current time and midnight, January 1, 1970 UTC. In addition, do you think value of time data type should avoid to use scientific notation?

Reply:
I understand this like that : 
The record type is FLOAT/Number.
This float can be used to store Integer, Float or Time, but at SenML level we have not this information.

At LWM2M level, thanks to LWM2M model we will be able to decode the float in float/integer or time.

Reply:
> In addition, do you think value of time data type should avoid to use scientific notation?

I don't think so. This is just notation, I can see any good reason to forbid it for now. Do you ?

Reply:
Agree.